flink-libraries/flink-state-processing-api/src/test/java/org/apache/flink/state/api/output/KeyedStateBootstrapOperatorTest.java

```diff
@@ -43,11 +48,53 @@
  */
 public class KeyedStateBootstrapOperatorTest {
 
-	private static final ValueStateDescriptor<Long> descriptor = new ValueStateDescriptor<Long>("state", Types.LONG);
+	private static final ValueStateDescriptor<Long> descriptor = new ValueStateDescriptor<>("state", Types.LONG);
+
+	private static final Long EVENT_TIMER = Long.MAX_VALUE - 1;
+
+	private static final Long PROC_TIMER = Long.MAX_VALUE - 2;
 
 	@Rule
 	public TemporaryFolder folder = new TemporaryFolder();
 
+	@Test
+	public void testTimerStateRestorable() throws Exception {
+		Path path = new Path(folder.newFolder().toURI());
+
+		OperatorSubtaskState state;
+		KeyedStateBootstrapOperator<Long, Long> bootstrapOperator = new KeyedStateBootstrapOperator<>(0L, path, new TimerBootstrapFunction());
+		try (KeyedOneInputStreamOperatorTestHarness<Long, Long, TaggedOperatorSubtaskState> harness = getHarness(bootstrapOperator)) {
+			harness.open();
+
+			harness.processElement(1L, 0L);
+			harness.processElement(2L, 0L);
+			harness.processElement(3L, 0L);
+			bootstrapOperator.endInput();
+
+			state = harness.extractOutputValues().get(0).state;
+		}
+
+		KeyedProcessOperator<Long, Long, Tuple3<Long, Long, TimeDomain>> procOperator = new KeyedProcessOperator<>(new SimpleProcessFunction());
+		try (KeyedOneInputStreamOperatorTestHarness<Long, Long, Tuple3<Long, Long, TimeDomain>> harness = getHarness(procOperator)) {
```

Comment:
Same here: could be nice to move this to a separate method, called say `assertHarnessOutputs(Harness<OUT>, OUT... )`

Reply:
👍